tools/cellprofiler/display_data_on_image.xml

```diff
@@ -0,0 +1,283 @@
+  f.write(FOURSPACES + "Display background image?:" + params['display_background_image'] + NEWLINE)
+  f.write(FOURSPACES + "Color map scale:" + color_map_scale + NEWLINE)
+  f.write(FOURSPACES + "Color map range:" + color_map_range + NEWLINE)
+  
+
+with open(input_pipeline) as fin:
+  lines = fin.readlines()
+  
+  k,v= lines[4].strip().split(':')
+
+  module_count = int(v)
+  new_count = module_count+1
+  lines[4]= k+":"+str(new_count)+"\n"
+  
+  with open("output","w") as f:
+    f.writelines(lines)
+    writemi()
+
+f.close() 
+        </configfile>
+    </configfiles>
+
+    <inputs>
+        <expand macro="input_pipeline_macro" />
+        <conditional name="con_display_obj_image">
+            <param name="display_obj_image" type="select" label="Display object or image measurements?">
+                <option value="Object">Object</option>
+                <option value="Image">Image</option>
+            </param>
+            <when value="Object">
+                <expand macro="display_object" />
+            </when>
+            <when value="Image">
+                <expand macro="display_image" />
+            </when>
+        </conditional>
+
+        <param name="display_background_image" label="Display background image?" type="select" display="radio">
+            <option value="Yes">Yes</option>
+            <option value="No">No</option>
+        </param>
+
+        <param name="image_to_display_measurements" label="Select the image on which to display the measurements" type="text"/>
+
+        <conditional name="con_display_mode">
+            <param name="display_mode" type="select" label="Display mode">
+                <option value="Text">Text</option>
+                <option value="Color">Color</option>
+            </param>
+            <when value="Text">
+                <param name="text_color" type="color" label="Text color" />
+                <param name="font_size" type="text" label="Font size(points)" />
+                <param name="number_of_decimals" type="text" label="Number of decimals" />
+                <param name="annotation_offset" type="text" label="Annotation offset (in pixels)" />
+            </when>
+            <when value="Color">
+                <param name="color_map" label="Color map" type="select">
+                    <option value="Default">Default</option>
+                </param>
+                <conditional name="con_color_map_scale">
+                    <param name="color_map_scale" label="Color map scale" type="select">
+                        <option value="Use this image's measurement range">Use this image's measurement range</option>
+                        <option value="Manual">Manual</option>
+                        <sanitizer sanitize="false"/>
+                    </param>        
+                    <when value="Manual">
+                        <param name="min_color_map_range" label="Min. color map range" type="text" />

```

Comment:
Can you please add a help here? What is this? How does it look like? Can we add a validator?